### PR TITLE
Improve log readability and allow control over what is logged

### PR DESCRIPTION
**Pull request by mrfelton**
Currently, the log output is messy and hard to follow.

This PR aims to improve the application logging with a more comprehensive and flexible logging
solution. We leverage https://github.com/appscot/debug-logger for this.

We use two separate log handlers to provide separation between electron logs (namespace=`zap:main`) and and lnd logs (namespace=`zap:lnd`). This provides a good level of control over what log out put you want to see. Some examples:

Only show log messages in the `zap:main` namespace
```
DEBUG=zap:main npm run dev
```

Only show log messages of level `warn` and above
```
DEBUG_LEVEL=warn npm run dev
```

The default configuration will show `debug` and above for all messages in the `zap:*` namespace.

**BEFORE:**
screenshot 2018-06-17 18 45 30

**AFTER:**
screenshot 2018-06-17 18 43 56

### CODE REVIEW

**Comment by Empact** [comment]
Could you rebase given the second commit is in?

**Comment by mrfelton** [comment]
I've rebased this from the updated master branch. Note that this branch also includes the commit from https://github.com/LN-Zap/zap-desktop/pull/433

**Comment by Empact** [comment]
Ah yeah thanks - could you rebase again now that #433 is in?

**Comment by mrfelton** [comment]
rebased
